db/index.js

```diff
@@ -0,0 +1,105 @@
+const mongoose = require('mongoose');
+const path = require('path');
+const fs = require('fs');
+const querystring = require('querystring');
+mongoose.connect('mongodb://localhost:/hackbnb');
+mongoose.promise = require('bluebird');
+const seed = '../data.json';
+const seedImages = '../images/'
+
+
+//listings service data schema.
+var similarListingSchema = mongoose.Schema({
+  //Listings Schema
+  id: Number,
+  lon: Number,
+  lat: Number,
+  city: String,
+  address: String,
+  desc: String,
+  price: Number,
+  ratings: Number,
+  photo_url: String,
+  photo: {
+    data: String,
+    contentType: String
+  }
+});
+
+//Fetches similar listings and returns.
+exports.fetchSimilarListings = function(roomId, cb) {
+  //Get room object
+  var room = similarListings.findOne({'id': roomId}, 'city address price ratings', function (err, obj) {
+    if (err) throw err;
+  });
+
+  var query = similarListings.find({'city': room.city});
+  query.select('id city address desc price ratings photo');
+  query.limit(6); //limit results to 6
+  query.sort({ratings: -1}); //sort by ratings.
+  
+  //execute query
+  query.exec((err, listings) => {
+    if (err) {
+      console.log('fetch err: ', err);
+      cb(err, null);
+    } else {
+      console.log('Fetch res: ', listings);
+      cb(null, listings);
+    }
+  });
+};
+
+
+//Store dummy images into db.
+exports.putSimilarListings = function(cb) {
+  seedDb(seed, cb);
+};
+
+
+var seedDb = function(content, cb) {
```

Comment:
I'd recommend this function to be in a separate file and script that is run independently.

Reply:
I just wanted to modularize all the code in one module. But I can break this part up in a separate file. I didn't really think it'll be that necessary.